### PR TITLE
Adding Security Context Constraints for OCP

### PR DESCRIPTION
**Pull request by dkosteck**
Providing SCC definitions needed for deployment on OCP

### CODE REVIEW

File: distros/openshift/security-context-constraints/scc-gitea-memcached-default.yaml
Comment (arora-sagar) [inline distros/openshift/security-context-constraints/scc-gitea-memcached-default.yaml:1]: @dkosteck I suppose would be good to have an empty line at the end of all the YAML files.

File: distros/openshift/security-context-constraints/scc-porch-system-porch-server.yaml
Comment (arora-sagar) [inline distros/openshift/security-context-constraints/scc-porch-system-porch-server.yaml:1]: @dkosteck I suppose you are missing an empty line at the end of the file.

File: distros/openshift/security-context-constraints/scc-config-management-monitoring-default.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+allowPrivilegeEscalation: false
+readOnlyRootFilesystem: true
+metadata:
+  name: scc-config-management-monitoring-default
+fsGroup:
+  type: MustRunAs
+  ranges:
+    - min: 2000
+      max: 2000
+groups:
```
Comment (arora-sagar) [inline distros/openshift/security-context-constraints/scc-config-management-monitoring-default.yaml:12]: @dkosteck Just a question why aren't you Dropping the capabilities here too? Like you did above `requiredDropCapabilities`

Comment (dkosteck) [reply distros/openshift/security-context-constraints/scc-config-management-monitoring-default.yaml:12]: @vjayaramrh Did we specifically look into this field, or was this not gleaned from the trials on OCP therefore it wasn't added?

File: distros/openshift/security-context-constraints/scc-porch-system-porch-server.yaml
Comment (dkosteck) [reply distros/openshift/security-context-constraints/scc-porch-system-porch-server.yaml:1]: This file actually had an empty line for EOF, the others did not, so I added them to make them consistent.

File: distros/openshift/security-context-constraints/scc-config-management-monitoring-default.yaml (hunk 1 above)
Comment (vjayaramrh) [reply distros/openshift/security-context-constraints/scc-config-management-monitoring-default.yaml:12]: We look at the KRM manifests to determine what capabilities needs to be added or dropped. In this case the

File: distros/openshift/security-context-constraints/scc-porch-system-porch-server.yaml
Comment (arora-sagar) [reply distros/openshift/security-context-constraints/scc-porch-system-porch-server.yaml:1]: First I commented on the others and then on this. Cool thanks

File: distros/openshift/security-context-constraints/scc-config-management-monitoring-default.yaml (hunk 1 above)
Comment (arora-sagar) [reply distros/openshift/security-context-constraints/scc-config-management-monitoring-default.yaml:12]: Okay, so are you adding in all of them? I suppose you can continue this good practice.

Comment (vjayaramrh) [reply distros/openshift/security-context-constraints/scc-config-management-monitoring-default.yaml:12]: FYI Not dropping all capabilities does not mean all capabilities would become available. Only capabilities that are specifically added will be available.

Comment (vjayaramrh) [reply distros/openshift/security-context-constraints/scc-config-management-monitoring-default.yaml:12]: For OCP, there would need to be SCC's created to match the capabilities requested.

Comment (arora-sagar) [reply distros/openshift/security-context-constraints/scc-config-management-monitoring-default.yaml:12]: Yes true, but still the process gets some basic capabilities even if we do not add capabilities. We can check that for each process via `capsh` command. 

But this is more about fine tunning `scc`. We can take this up in another MR.

**Comment by arora-sagar** [comment]
Sorry I just realized, you are merging it in `V3.0.0`?  Technically the functionality was added after `v3.0.0` release so we should add it in `main` branch.

File: distros/openshift/security-context-constraints/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# distros/openshift/security-context-constraints
+
+## Security Context Constraints
+
+OpenShift deployments require Security Context Constraints to be applied for containers to function properly. The ones defined here were identified through testing the deployment against an OpenShift cluster and creating SCCs. These are prerequisites, and should be applied before attempting to install Nephio components.
```
Comment (efiacor) [inline distros/openshift/security-context-constraints/README.md:5]: Do we need to add an entry to the docs for these prerequisites?
https://docs.nephio.org/docs/guides/install-guides/install-on-openshift/

File: distros/openshift/security-context-constraints/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+## Storage
+
+Gitea requires storage to be configured on the OpenShift cluster. Whether LVM, Local Storage, etc. there needs to be storage and a default Storage Class for the Persistent Volumes to be allocated.
```
Comment (efiacor) [inline distros/openshift/security-context-constraints/README.md:9]: Is this part of security constraints or should it be mentioned in the overall guide somewhere? With examples on how to?

**Comment by dkosteck** [comment]
> Sorry I just realized, you are merging it in `V3.0.0`? Technically the functionality was added after `v3.0.0` release so we should add it in `main` branch.

@arora-sagar So if you look at the previous PR Vish pushed, it was merged to both v3.0.0 and main, since this testing was done with the v3.0.0 release: https://github.com/nephio-project/catalog/pull/73 and https://github.com/nephio-project/catalog/pull/74

File: distros/openshift/security-context-constraints/README.md (hunk 2 above)
Comment (dkosteck) [reply distros/openshift/security-context-constraints/README.md:5]: I do think documenting this in the docs on a following PR would benefit readers, yes

File: distros/openshift/security-context-constraints/README.md (hunk 3 above)
Comment (dkosteck) [reply distros/openshift/security-context-constraints/README.md:9]: Yes, I think checking the docs for this content and making sure they reflect this in a follow up PR would be good

**Comment by arora-sagar** [comment]
> > Sorry I just realized, you are merging it in `V3.0.0`? Technically the functionality was added after `v3.0.0` release so we should add it in `main` branch.
> 
> @arora-sagar So if you look at the previous PR Vish pushed, it was merged to both v3.0.0 and main, since this testing was done with the v3.0.0 release: #73 and #74

Yeah true, I realized later but then it means we are back pushing the code. We should also sync the `main` branch because that will be used for R4 or R3.XX (if will we do an intermediate release)?

File: distros/openshift/security-context-constraints/scc-config-management-monitoring-default.yaml (hunk 1 above)
Comment (vjayaramrh) [reply distros/openshift/security-context-constraints/scc-config-management-monitoring-default.yaml:12]: @arora-sagar Great points, additionally, we may want to look at firming up the securityContext sections of the pods/containers in the KRM files as well.

File: distros/openshift/security-context-constraints/README.md (hunk 2 above)
Comment (efiacor) [reply distros/openshift/security-context-constraints/README.md:5]: https://github.com/nephio-project/nephio/issues/801

File: distros/openshift/security-context-constraints/README.md (hunk 3 above)
Comment (efiacor) [reply distros/openshift/security-context-constraints/README.md:9]: https://github.com/nephio-project/nephio/issues/801

**Comment by efiacor** [comment]
Needs to be cherrypicked to main branch

**Pull request merged**